index.bs

```diff
@@ -2829,31 +2898,64 @@ an assertion. Its {{PublicKeyCredentialRequestOptions/challenge}} member MUST be
         The value is treated as a hint, and MAY be overridden by the [=client=].
 
     :   <dfn>rpId</dfn>
-    ::  This OPTIONAL member specifies the [=relying party identifier=] claimed by the caller. If omitted, its value will
+    ::  This OPTIONAL member specifies the [=RP ID=] claimed by the [=[RP]=].
+        The [=client=] MUST verify that the [=[RP]=]'s [=origin=] matches the [=scope=] of this [=RP ID=].
+        The [=authenticator=] MUST verify
+        that this [=RP ID=] exactly equals the [=rpId=] of the [=credential=] to be used for the [=authentication ceremony=].
+
+        If not specified, its value will
         be the {{CredentialsContainer}} object's [=relevant settings object=]'s [=environment settings object/origin=]'s
         [=effective domain=].
 
     :   <dfn>allowCredentials</dfn>
-    ::  This OPTIONAL member contains a list of {{PublicKeyCredentialDescriptor}} objects representing [=public key credentials=]
-        acceptable to the caller, in descending order of the caller's preference (the first item in the list is the most
-        preferred credential, and so on down the list).
+    ::  This OPTIONAL member is used by the [=client=] to find [=authenticators=] eligible for this [=authentication ceremony=].
+        It can be used in two ways:
+
+          - If the [=user account=] to authenticate is already identified (e.g., if the user has entered a username),
+            then the [=[RP]=] SHOULD use this member to list the [=credentials=] [=registration ceremony|registered=]
+            to the [=user account=].
+            This SHOULD usually include all of the [=user account's=] credentials.
+
+            The [=list/items=] SHOULD specify {{PublicKeyCredentialDescriptor/transports}} whenever possible.
+            This helps the [=client=] optimize the user experience for any given situation.
+            Also note that the [=[RP]=] does not need to filter the list when requesting [=user verification=] &mdash;
+            the [=client=] will automatically ignore non-eligible credentials
+            if {{PublicKeyCredentialRequestOptions/userVerification}} is set to {{UserVerificationRequirement/required}}.
+
+          - If the [=user account=] to authenticate is not already identified,
+            then the [=[RP]=] MAY leave this member [=list/empty=] or unspecified.
+            In this case, only [=discoverable credentials=] will be utilized in this [=authentication ceremony=],
+            and the [=user account=] MAY be identified by the {{AuthenticatorAssertionResponse/userHandle}}
+            of the resulting {{AuthenticatorAssertionResponse}}.
+            If the available [=authenticators=] [=contain=] more than one [=discoverable credential=] [=scoped=] to the [=[RP]=],
+            the credentials are displayed for the user to select from (see step 7 of [[#sctn-op-get-assertion]]).
```

Comment:
Hm, almost. It could also be done by the authenticator itself (in the case of CaBLE for example). We do have the term [client side](https://w3c.github.io/webauthn/#client-side) defined as "[...] the combination of the user’s client platform, authenticators, and everything gluing it all together", but I can't think of a way to use that without weighing everything down too much. Do you have any ideas?

Reply:
How about this?
```suggestion
            the credentials are displayed, by the [=client platform=] (or in some cases the [=authenticator=] itself) for the user to select from (see step 7 of [[#sctn-op-get-assertion]]).
```

Reply:
Hah, I guess it didn't need to be so difficult after all. Thanks! :smile: I actually simplified it even a little more.